### PR TITLE
Security updates 2022-10-04

### PR DESCRIPTION
**Pull request by tlbdk**
# Nov 3 2022 Security Releases

## X.509 Email Address 4-byte Buffer Overflow (High) (CVE-2022-3602)
A buffer overrun can be triggered in X.509 certificate verification, specifically in name constraint checking. Note that this occurs after certificate chain signature verification and requires either a CA to have signed the malicious certificate or for the application to continue certificate verification despite failure to construct a path to a trusted issuer. An attacker can craft a malicious email address to overflow four attacker-controlled bytes on the stack. This buffer overflow could result in a crash (causing a denial of service) or potentially remote code execution.

Impacts:
* All versions of the v18.x and v19.x releases lines.

## X.509 Email Address Variable Length Buffer Overflow (High) (CVE-2022-3786)
A buffer overrun can be triggered in X.509 certificate verification, specifically in name constraint checking. Note that this occurs after certificate chain signature verification and requires either a CA to have signed a malicious certificate or for an application to continue certificate verification despite failure to construct a path to a trusted issuer. An attacker can craft a malicious email address in a certificate to overflow an arbitrary number of bytes containing the . character (decimal 46) on the stack. This buffer overflow could result in a crash (causing a denial of service).

In a TLS client, this can be triggered by connecting to a malicious server. In a TLS server, this can be triggered if the server requests client authentication and a malicious client connects.

OpenSSL versions 3.0.0 to 3.0.6 are vulnerable to this issue.

Impacts:
* All versions of the v18.x and v19.x releases lines.

## DNS rebinding in --inspect via invalid octal IP address (Medium) (CVE-2022-43548)
The Node.js rebinding protector for --inspect still allows invalid IP address, specifically, the octal format. An example of an octal IP address is 1.09.0.0, the 09 octet is invalid because 9 is not a number in the base 8 number system. Browsers such as Firefox (tested on latest version m105) will still attempt to resolve this invalid octal address via DNS. When combined with an active --inspect session, such as when using VSCode, an attacker can perform DNS rebinding and execute arbitrary code

Thank you to @haxatron1 for reporting this vulnerability.

Impacts:
* All versions of the v14.x, v16.x, v18.x, and v19.x releases lines.

### CODE REVIEW

**Comment by tlbdk** [comment]
This can wait until monday

File: build-all.sh [hunk 1]
```diff
@@ -3,9 +3,9 @@
 set -eu
 
 NODE_STABLE="16"
-NODE_VERSIONS=${NODE_VERSIONS:="18.10.0 16.17.1 14.20.1 12.22.12"}
+NODE_VERSIONS=${NODE_VERSIONS:="18.12.1 16.18.1 14.21.1"}
```
Comment (mex) [inline build-all.sh:6]: They're still using Node 12 in [firmware](https://github.com/connectedcars/firmware/blob/master/Dockerfile)

**Review by mex — COMMENTED** [review]
Looks like the build is failing.

File: build-all.sh (hunk 1 above)
Comment (tlbdk) [reply build-all.sh:6]: Node is not updating 12.x anymore so don't know if it makes sense to keep building it.

Comment (mex) [reply build-all.sh:6]: Makes sense not to build it as the old version will stay in the container registry.

**Review by mex — APPROVED** [review]

**Pull request merged**